### PR TITLE
feat(litmus): add hpa

### PR DESCRIPTION
**Pull request by itninja-hue**
#### What this PR does / why we need it:
Adds HPA for authsever , frontend and server deployment.
enable autoscaling for load peaks.
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] DCO signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by Jasstkn** [comment]
@itninja-hue hi. please, take a look into failed jobs.

**Review by Jasstkn — APPROVED** [review]
LGTM

File: charts/litmus/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 appVersion: "2.11.0"
 description: A Helm chart to install ChaosCenter
 name: litmus
-version: 2.12.0
+version: 2.12.1
```
Comment (aslafy-z) [inline charts/litmus/Chart.yaml:5]: @Jasstkn should the minor be bumped in here? Or patch is ok?

**Review by aslafy-z — APPROVED** [review]
LGTM

**Comment by Jasstkn** [comment]
> LGTM 

Patch is fine by me.

File: charts/litmus/README.md [hunk 2]
```diff
@@ -126,6 +137,11 @@ $ helm install litmus-portal litmuschaos/litmus
 | portal.server.authServer.securityContext.runAsUser | int | `2000` |  |
 | portal.server.authServer.volumeMounts | list | `[]` |  |
 | portal.server.authServer.volumes | list | `[]` |  |
+| portal.server.autoscaling.enabled | bool | `false` |  |
+| portal.server.autoscaling.maxReplicas | int | `3` |  |
+| portal.server.autoscaling.minReplicas | int | `2` |  |
+| portal.server.autoscaling.targetCPUUtilizationPercentage | int | `50` |  |
+| portal.server.autoscaling.targetMemoryUtilizationPercentage | int | `50` |  |
```
Comment (gdsoumya) [inline charts/litmus/README.md:144]: all graphql config needs to be under `graphqlServer` for consistency, it seems other existing configs for gql server is also not grouped together properly, we should surely clean this up sometime @Jasstkn @imrajdas @Jonsy13

**Review by gdsoumya — APPROVED** [review]

File: charts/litmus/README.md (hunk 2 above)
Comment (Jasstkn) [reply charts/litmus/README.md:144]: Yes, nice catch tho. I'll create an issue to figure out a better structure for our components.

File: charts/litmus/templates/server-hpa.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{{- if .Values.portal.server.autoscaling.enabled -}}
```
Comment (imrajdas) [inline charts/litmus/templates/server-hpa.yaml:1]: gql server can't be horizontal scale. So, we can remove this spec

File: charts/litmus/values.yaml [hunk 4]
```diff
@@ -129,6 +135,12 @@ portal:
 
   server:
     replicas: 1
+    autoscaling:
+      enabled: false
+      minReplicas: 2
+      maxReplicas: 3
```
Comment (imrajdas) [inline charts/litmus/values.yaml:141]: same:
gql server can't be horizontal scale. So, we can remove this spec

**Review by imrajdas — CHANGES_REQUESTED** [review]

File: charts/litmus/templates/server-hpa.yaml (hunk 3 above)
Comment (itninja-hue) [reply charts/litmus/templates/server-hpa.yaml:1]: how come ?  
the ability to change the number of replicas is present on stable (master) 
 https://github.com/litmuschaos/litmus-helm/blob/master/charts/litmus/values.yaml#L144 
 https://github.com/litmuschaos/litmus-helm/blob/master/charts/litmus/templates/server-deployment.yaml#L13 
 
 Please can you give more details  :pray:

**Comment by aslafy-z** [comment]
@itninja-hue Please rebase and bump the Chart version.

**Comment by itninja-hue** [comment]
@Jasstkn  @imrajdas  @gdsoumya  ping i see the pipe failing on azure due to  
```Error
/usr/local/bin/kind: line 1: Error: command not foun
```

**Comment by Jasstkn** [comment]
@imrajdas hey. can you give more detailed feedback if this change is valid or not?

**Comment by Jonsy13** [comment]
Added @Adarshkumar14 for review

**Comment by Jonsy13** [comment]
Also resolve conflicts in this PR.

**Review by Adarshkumar14 — APPROVED** [review]
LGTM

**Review by Jasstkn — APPROVED** [review]

**Review by Jasstkn — CHANGES_REQUESTED** [review]
Hey. Everything looks great - just a small note, could you bump only patch version of chart? we have an agreement to follow the litmus version here and all changes should be done via patch version.

I'll add a note about it in contribution guide. Thanks in advance!

**Comment by Jasstkn** [comment]
@itninja-hue thanks again for your contribution!

**Pull request merged**